main.tf

```
@@ -866,3 +866,63 @@ resource "juju_integration" "ldap-to-keystone" {
     endpoint = "domain-config"
   }
 }
+
+resource "juju_application" "tempest" {
+  count = var.enable-validation ? 1 : 0
+  name  = "tempest"
+  model = juju_model.sunbeam.name
+
+  charm {
+    name     = "tempest-k8s"
+    channel  = var.tempest-channel
```

Like other charms defined, I think we should set the default channel to openstack channel if not specified?
```suggestion
    channel = var.tempest-channel == null ? var.openstack-channel : var.tempest-channel
```

---

Done